### PR TITLE
[work in progress] Rework samples

### PR DESCRIPTION
**Pull request by kuntzed**
To improve the readability of the samples applications we do some small refactorings and bugfixes. This includes implementation and configuration harmonisations. Furthermore, additional examples for k8b are added and the instructions (README files) for the samples are rewritten.

### CODE REVIEW

File: samples/java-security-usage/README.md
Comment (finkmanAtSap) [inline samples/java-security-usage/README.md:1]: Nice! This should be much smoother now for customers.

**Review by finkmanAtSap — APPROVED** [review]

**Review by finkmanAtSap — CHANGES_REQUESTED** [review]
I think the E2E tests might be failing in the Github actions workflow because the `maven.compiler.target` or `maven.compiler.source` option has been replaced by `maven.compiler.release`?

File: samples/java-security-usage/README.md [hunk 1]
```diff
@@ -1,140 +1,164 @@
...
-cf create-service xsuaa application xsuaa-java-security -c xs-security.json
+kubectl apply -f k8s/deployment.yml
 ```
+</details>
+
+### 3. Assign Role Collection to user
+> :bulb: You can postpone this step if you first want to test the application without the required authorization.
+
+To get full access to the sample application, you need a user having the role collection `Sample Viewer (java-security-usage)` assigned.
+This can be done in the SAP BTP Cockpit or using the btp CLI.
```
[the remark this answers is not in the document]
Comment (kuntzed) [reply samples/java-security-usage/README.md:61]: There are two official names for this CLI:
1. SAP BTP Command Line Interface
2. btp CLI

File: docs/HowToFetchToken.md
[the remark this answers is not in the document]
Comment (kuntzed) [reply docs/HowToFetchToken.md:1]: I would prefer not to force users to pull together the instructions from several places. But I wonder if we can remove this postman collection stuff. Part of the instructions require the command-line anyway. I see two options:
1. Reduce the instructions two the curl commands and additionally provide a link to the official documentation
2. Remove the complete file and just add the required commands directly to the README files.

What do you think?

File: samples/java-security-usage/README.md [hunk 2]
```diff
@@ -1,140 +1,164 @@
...
+To assign the role collection of the sample application to a user you have basically two options:
+
+1. Navigate to the user by clicking on `Security` -> `Users`,
+select the user and click on `Assign Role Collection`
+(more info at [help.sap.com](https://help.sap.com/docs/btp/sap-business-technology-platform/find-users-and-their-role-collection-assignments)).
+2. Navigate to the role collection by clicking on `Security` -> `Role Collections`,
+select `Sample Viewer (java-security-usage)`,
+click on `Edit` to add the user and finish by clicking on `Save`
+(more info at [help.sap.com](https://help.sap.com/docs/btp/sap-business-technology-platform/assign-users-to-role-collections)).
+</details>
```
[the remark this answers is not in the document]
Comment (kuntzed) [reply samples/java-security-usage/README.md:75]: I prefer self-contained instructions. If you really have a strong opinion here, then I could remove the detailed instructions. But I would like to keep mentioning the two ways of assigning the role collection.

File: samples/java-security-usage/README.md
Comment (kuntzed) [reply samples/java-security-usage/README.md:1]: Thanks.

File: samples/java-security-usage/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (finkmanAtSap) [reply samples/java-security-usage/README.md:61]: Aaaah, I see. Thought it was a typo. Nevermind then if it's the official name.

File: samples/java-security-usage/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (finkmanAtSap) [reply samples/java-security-usage/README.md:75]: We can leave it in but I don' think we'll manage to maintain it if something changes in the UI.

[the remark this answers is not in the document]
Comment (kuntzed) [reply samples/java-security-usage/README.md:75]: That's true. But besides that not being very likely, we have the links to help.sap.com in place as a fallback.

File: docs/HowToFetchToken.md
[the remark this answers is not in the document]
Comment (finkmanAtSap) [reply docs/HowToFetchToken.md:1]: Option 1 sounds good to me if you can work in the OTP code into the curl scripts as required.

[the remark this answers is not in the document]
Comment (kuntzed) [reply docs/HowToFetchToken.md:1]: Ok, will do

**Review by finkmanAtSap — APPROVED** [review]

**Pull request merged**